### PR TITLE
key files

### PR DESCRIPTION
**Pull request by nakonn**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
+variable "key_pair_permission" {
+  description = "Permission to set for the private key file."  
+  type        = list(number)
+}
+
+variable "private_key_path" {
+  description = "Path to the private key file."
```
Comment (Jiraiya106) [inline variables.tf:8]: sensitive

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,29 @@
+variable "key_pair_permission" {
+  description = "Permission to set for the private key file."
```
Comment (Jiraiya106) [inline variables.tf:2]: sensitive

File: variables.tf [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+   type        = list(string)
+}
+
+variable "key_name" {
+  description = "Name of the key pair."  
+   type        = list(string)
+}
+
+variable "private_key_algorithm" {
+  description = "Algorithm used for generating the private key."
```
Comment (Jiraiya106) [inline variables.tf:24]: sensitive

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+   type        = list(string)
+}
+
+variable "private_key_algorithm" {
+  description = "Algorithm used for generating the private key."  
+   type        = list(string)
+}
+
+variable "private_key_rsa_bits" {
+  description = "the size of the generated RSA key, in bits"
```
Comment (Jiraiya106) [inline variables.tf:30]: sensitive

File: variables.tf (hunk 1 above)
Comment (nakonn) [reply variables.tf:8]: fixed

File: variables.tf (hunk 2 above)
Comment (nakonn) [reply variables.tf:2]: fixed

File: variables.tf (hunk 3 above)
Comment (nakonn) [reply variables.tf:24]: fixed

File: variables.tf (hunk 4 above)
Comment (nakonn) [reply variables.tf:30]: fixed

**Pull request merged**